### ISSUE TITLE
[HybridApp] - Push notification duplicated and not styled

### ISSUE THREAD

**Issue by AndrewGable** [Expensify/App#46837]
## Problem

@roryabraham and I were DM-ing and noticed the two bugs with push notifications on iOS HybridApp:
1. All messages are duplicated
2. We do not have the user's name like we do on NewDot standalone app, it just says "Expensify"
3. HybridApp isn't using our custom notifications sounds from antfood

<img src='https://github.com/user-attachments/assets/4217ef0c-b69e-452f-a7b7-bfb804e29039' width='250' />

## Solution
Fix it!

cc @Julesssss @staszekscp

**Comment by AndrewGable** [comment]
Additionally, @quinthar was receiving push notifications in a "stack" and not in the "chat" style that I believe we use on New Expensify standalone app.


https://github.com/user-attachments/assets/2aca7d42-dbff-4895-9bd5-6d006903cb67

**Comment by Julesssss** [comment]
> All messages are duplicated

This is being tracked separately [here](https://github.com/Expensify/App/issues/46485) 👍 

> We do not have the user's name like we do on NewDot standalone app, it just says "Expensify"

Pretty sure we simply need to 'build' the notification locally and group them to the same ID (reportID IIRC) like we do [here](https://github.com/Expensify/App/blob/6db3f487c899ce4ddbacf3526cad27a4b27ab7b0/android/app/src/main/java/com/expensify/chat/customairshipextender/CustomNotificationProvider.java#L167) for NewDot.

**Comment by Julesssss** [comment]
NewDot notifications aren't being styled as we are not bundling the Onyx data into the Airship data package. This is due to [backend logic](https://github.com/Expensify/Web-Expensify/blob/14a653ea8e8f149b9df2d13b40b750e623b53860/lib/MobilePushNotifications.php#L63-L96) which applies data based on the intended application. 

Now that Hybrid receieves notifications for both NewDot and OldDot I need to route notifications based on NVP, not intended application. This is related to the changes that need to be made [here](https://github.com/Expensify/App/issues/46485) to reduce duplicated notification delivery.

**Comment by Julesssss** [comment]
Starting with the grouped notifications here: https://github.com/Expensify/Mobile-Expensify/pull/12922

**Comment by Julesssss** [comment]
**Today:** 
- Fixed the Android notification group issue the need to be swiped away 1-by-1, but this is not possible to test/verify (uninstalling the prod app to test my build also resolves the issue)
- Made progress on dupe notifications [issue](https://github.com/Expensify/App/issues/46485) 
- Holding the style changes for now -- the changes I'm making for dupe notifications are a prerequisite

**Comment by roryabraham** [comment]
Hey @Julesssss, another thing I noticed is that HybridApp isn't using our custom notifications sounds from antfood

**Comment by Julesssss** [comment]
Ah thanks, I'll keep that lower priority for now.

**Comment by Julesssss** [comment]
- Discussions [here](https://github.com/Expensify/Web-Expensify/pull/43039/files#r1713991837) about whether to bundle the style changes into this PR, I think I will probably end up doing this
- Here are some new problems that arise after these changes, I have solutions but need to make sure we are all aligned. I hope to have a quick chat with Andrew today about this, if not it will be early next week

**Comment by Julesssss** [comment]
PR for dupe notifications should be merged very shortly

I think this is what we need to do for the incorrectly styled Hybrid NewExperience notifications:

<img width="784" alt="Screenshot 2024-08-12 at 17 54 25" src="https://github.com/user-attachments/assets/0124bd4f-3187-49fc-9be2-fedc65bc4fca">

**Comment by Julesssss** [comment]
1) update the Airship request data [here](https://github.com/Expensify/Web-Expensify/blob/a2a4e122b9aa0f91aa5746c1b5b63e08757c965f/lib/UrbanAirshipAPI.php#L133) -- easy because once my other PR is merged the accounts have already been split
2) Update the pushData [here](https://github.com/Expensify/Web-Expensify/blob/a2a4e122b9aa0f91aa5746c1b5b63e08757c965f/lib/MobilePushNotifications.php#L63-L77) and [here](https://github.com/Expensify/Web-Expensify/blob/a2a4e122b9aa0f91aa5746c1b5b63e08757c965f/lib/MobilePushNotifications.php#L82-L96) -- difficult because the account sorting occurs later on in UrbanAirshipAPI::sendPushNotification
3) Maybe change that boolean set 'app' variable, I am not sure yet(edited)

**Comment by Julesssss** [comment]
I'm back from OOO. 

We didn't merge the fix last week as it is unperforment. Today I am working on an auth performance improvement which will unblock the previous web PR.

**Comment by Julesssss** [comment]
Auth [PR](https://github.com/Expensify/Auth/pull/12128) is in progress for the dupe notifications issue. This makes the Web-E changes performant.

**Comment by Julesssss** [comment]
First of the two dupe notification PR's merged. Next up is [this Web-E PR](https://github.com/Expensify/Web-Expensify/pull/43039)

**Comment by Julesssss** [comment]
[Recent PR merges](https://github.com/Expensify/App/issues/46485#issuecomment-2304678789) should have resolved issues 1 & 2.

**Comment by Julesssss** [comment]
> Additionally, @quinthar was receiving push notifications in a "stack" and not in the "chat" style that I believe we use on New Expensify standalone app.

[This PR](https://github.com/Expensify/Mobile-Expensify/pull/12922) should resolve the non-grouping issue.

**Comment by Julesssss** [comment]
> Hey @Julesssss, another thing I noticed is that HybridApp isn't using our custom notifications sounds from antfood

Splitting this into a [new issue ](https://github.com/Expensify/App/issues/47868)to avoid confusion amongst all these issues.
